src/_pytest/fixtures.py

```diff
@@ -1502,11 +1475,28 @@ def getfixtureinfo(
         usefixtures = tuple(
             arg for mark in node.iter_markers(name="usefixtures") for arg in mark.args
         )
-        initialnames = usefixtures + argnames
-        initialnames, names_closure, arg2fixturedefs = self.getfixtureclosure(
-            initialnames, node, ignore_args=_get_direct_parametrize_args(node)
+        initialnames = cast(
+            Tuple[str],
+            tuple(
+                dict.fromkeys(
+                    tuple(self._getautousenames(node.nodeid)) + usefixtures + argnames
+                )
+            ),
+        )
```

Comment:
Are you using `dict.fromkeys` here to deduplicate the tuple while preserving order?

If so let's create a local function to improve readability:



```suggestion
        def deduplicate_names(seq: Iterable[str]) -> Tuple[str, ...]:
            """De-duplicate the list of names while keeping the original order."""
            # Ideally we would use a set, but it does not preserve insertion order.
            return tuple(dict.fromkeys(seq))

        initialnames = deduplicate_names(tuple(self._getautousenames(node.nodeid)) + usefixtures + argnames)
```